### PR TITLE
CPA-89 Reduce HealthKit Coupling

### PR DESCRIPTION
**Pull request by ps2**
https://tidepool.atlassian.net/browse/CPA-89

* Updates for healthkit decoupling: https://github.com/tidepool-org/LoopKit/pull/508
* [Use MKRingProgressView from swift package](https://github.com/tidepool-org/Loop/commit/7ce092f9eb4284726c93c1ac89ac819487f8e89c)
* [Update to ZipFoundation for providing zip file creation abilities](https://github.com/tidepool-org/Loop/commit/3b92d56c259fd00d6a860f7c73bd9f9b2403c734)

### CODE REVIEW

File: Loop/Managers/Alerts/AlertManager.swift [hunk 1]
```diff
@@ -799,7 +799,7 @@ extension AlertManager {
     }
     
     func presentCouldNotResetLoopAlert(error: Error) {
-        let alert = UIAlertController(title: "Could Not Restart Tidepool Loop", message: "While trying to restart Tidepool Loop an error occured.\n\n\(error.localizedDescription)", preferredStyle: .alert)
+        let alert = UIAlertController(title: "Could Not Restart Loop", message: "While trying to restart Loop an error occured.\n\n\(error.localizedDescription)", preferredStyle: .alert)
```
Comment (nhamming) [inline Loop/Managers/Alerts/AlertManager.swift]: appName?

File: Loop/Managers/DeviceDataManager.swift [hunk 2]
```diff
@@ -180,35 +180,35 @@ final class DeviceDataManager {
...
-        carbStore.sharingDenied ||
-        doseStore.sharingDenied ||
-        sleepDataSharingDenied
-    }
+//    private var sharingDenied: Bool {
+//        return healthStore.authorizationStatus(for: HealthKitSampleStore.glucoseType) == .sharingDenied ||
+//        carbStore.sharingDenied ||
+//        doseStore.sharingDenied ||
+//        sleepDataSharingDenied
+//    }
```
Comment (nhamming) [inline Loop/Managers/DeviceDataManager.swift]: remove?

File: Loop/Managers/DeviceDataManager.swift [hunk 3]
```diff
@@ -334,7 +350,7 @@ final class DeviceDataManager {
         self.automaticDosingStatus = automaticDosingStatus
 
         // HealthStorePreferredGlucoseUnitDidChange will be notified once the user completes the health access form. Set to .milligramsPerDeciliter until then
-        displayGlucoseUnitObservable = DisplayGlucoseUnitObservable(displayGlucoseUnit: glucoseStore.preferredUnit ?? .milligramsPerDeciliter)
+        displayGlucoseUnitObservable = DisplayGlucoseUnitObservable(displayGlucoseUnit: .milligramsPerDeciliter)
```
Comment (nhamming) [inline Loop/Managers/DeviceDataManager.swift:344]: why not use the user preferred unit?

File: Loop/Views/HowMuteAlertWorkView.swift [hunk 4]
```diff
@@ -32,7 +32,7 @@ When using Mute Alerts, also consider the impact of using iOS Focus Modes.
                                 .background(guidanceColors.warning)
                                 .clipShape(RoundedRectangle(cornerRadius: 5, style: .continuous))
                             
-                            Text(NSLocalizedString("Tidepool Loop Mute Alerts", comment: "Section title for description that mute alerts is temporary"))
+                            Text(NSLocalizedString("Loop Mute Alerts", comment: "Section title for description that mute alerts is temporary"))
```
Comment (nhamming) [inline Loop/Views/HowMuteAlertWorkView.swift]: appName?

File: Loop/Views/HowMuteAlertWorkView.swift [hunk 5]
```diff
@@ -52,7 +52,7 @@ After the mute period ends, your alert sounds will resume.
                             Text(NSLocalizedString("iOS Focus Mode", comment: "Section title for description of how mute alerts work with focus mode"))
                                 .bold()
                         }
-                        Text(NSLocalizedString("If iOS Focus Mode is ON and Mute Alerts is OFF, Critical Alerts will still be delivered, but non-Critical Alerts will be silenced until Tidepool Loop is added to each Focus mode as an Allowed App.", comment: "Description of how mute alerts works with focus mode"))
+                        Text(NSLocalizedString("If iOS Focus Mode is ON and Mute Alerts is OFF, Critical Alerts will still be delivered, but non-Critical Alerts will be silenced until Loop is added to each Focus mode as an Allowed App.", comment: "Description of how mute alerts works with focus mode"))
```
Comment (nhamming) [inline Loop/Views/HowMuteAlertWorkView.swift]: appName?

File: Loop/Managers/Alerts/AlertManager.swift (hunk 1 above)
Comment (ps2) [reply Loop/Managers/Alerts/AlertManager.swift]: This is non user facing, for Tidepool Loop, so I thought Loop might work here. But I can update it to string formatting, too.  This is something that would've been nice to catch on the initial review when "Tidepool Loop" landed in shared code.

File: Loop/Managers/DeviceDataManager.swift (hunk 3 above)
Comment (ps2) [reply Loop/Managers/DeviceDataManager.swift:344]: The HealthKit preferred lookup is an asynchronous call, and is generally not available at this point in app initialization. The correct value will be propagated on completion of HK authorization and glucose store initialization.

File: Loop/Views/HowMuteAlertWorkView.swift (hunk 4 above)
Comment (ps2) [reply Loop/Views/HowMuteAlertWorkView.swift]: 👍 Another one that would've been nice to have caught in the initial commit.

**Review by nhamming — APPROVED** [review]
LGTM

**Pull request merged**